### PR TITLE
Card Grant UI for Organizers + Refactor Stripe Card to partials

### PR DESCRIPTION
**Pull request by garyhtou**
## Summary of the problem

The UI for organizers to view and manage individual card grants needed some work. Organizers only had access to the Stripe Card show page and not the Card Grant show page.

## Describe your changes

As a part of these changes, I needed to refactor the 200+ line Stripe Card show page into partials (it's now 30 lines, yay!). this probably should have been 2 PRs. There were no visual changes to the Stripe Card show page except:

|Before|After|
|-|-|
| ||

### CardGrant Show Page for Grantee (the one receiving the grant)
NO VISUAL CHANGES
image

### CardGrant Show Page for Grantor (an organizer on the event)
image

image
Oh look! you can cancel and freeze/unfeeze a card

### CardGrant Show Page for Admins
image

image

---

### Event's Card Overview
no longer lists cards from Grants

|Before|After|
|-|-|
| *trust me* there were 7 pages of cards 😵‍💫 ||

---

### Other notable changes
- If a Stripe Card belongs to a grant, it's show page will redirect to the Grant's show page (for everyone except admins)

### CODE REVIEW

**Comment by garyhtou** [comment]
i'd love to get some eyes on this as a sanity check

**Comment by sampoder** [comment]
Re: cards on the overview page, https://github.com/hackclub/hcb/issues/4391

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -270,7 +270,8 @@ def emburse_card_overview
   end
 
   def card_overview
-    @stripe_cards = @event.stripe_cards.includes(:stripe_cardholder, :user).order("created_at desc")
+    @stripe_cards = @event.stripe_cards.where.missing(:card_grant)
+                          .includes(:stripe_cardholder, :user).order("created_at desc")
```
Comment (sampoder) [inline app/controllers/events_controller.rb]:
```suggestion
                          .includes(:stripe_cardholder, :user)
                          .order(
                            Arel.sql("subledger_id IS NULL DESC"),
                            "created_at desc"
                          )
```

Thoughts on instead placing the grant cards after those that aren't grant cards?

**Comment by garyhtou** [comment]
@maxwofford do you have context on https://github.com/hackclub/hcb/issues/4391 and why Organizers prefer to have cards from `CardGrant`s to show on the Cards Overview page?

I feel like they should be kept separate. Org's cards should be accessible on the Cards Overview page and `CardGrant` cards should be accessible from their grants (on the Transfers page)

**Comment by garyhtou** [comment]
> @/maxwofford do you have context on #4391 and why Organizers prefer to have cards from `CardGrant`s to show on the Cards Overview page?

After chatting with Jasper today, we'll be removing the from the Cards page

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (garyhtou) [reply app/controllers/events_controller.rb]: https://github.com/hackclub/hcb/pull/4940#issuecomment-1894452478

**Pull request merged**